src/session/session_helper.c

```diff
@@ -169,10 +169,9 @@ __wt_session_set_last_error(
      * Load error codes and message into err_info. If the message is empty or is NULL (indicating
      * success), use static string buffers. Otherwise, format the message into the buffer.
      *
-     * If err == 0, either: we are opening the session and err_msg should be initialized to
-     * WT_ERROR_INFO_EMPTY; or we are at the start of an API call, in which case fmt should be NULL
-     * and err_msg should be set to WT_ERROR_INFO_SUCCESS. NULL implying success here saves us a
-     * strcmp to validate that we never set err = 0 with a custom message.
+     * When err is 0, it signifies the start of an API call. The fmt needs to be set to NULL, which
+     * sets the err_msg to WT_ERROR_INFO_SUCCESS. The NULL here implies success saving us from an
+     * extra strcmp().
```

Comment:
I think this comment could be made clearer.

Reply:
I have tried to make this clearer. Let me know if it is good now.